k8s/migration/internal/controller/migrationplan_controller.go

```diff
@@ -1454,28 +1455,54 @@ func (r *MigrationPlanReconciler) migrateRDMdisks(ctx context.Context, migration
 			log.FromContext(ctx).Info("Skipping update for RDMDisk CR as it is already being managed or in error state", "rdmDiskName", rdmDiskCR.Name, "phase", rdmDiskCR.Status.Phase)
 			continue
 		}
-		rdmDisk := &vjailbreakv1alpha1.RDMDisk{}
-		err := r.Get(ctx, types.NamespacedName{
-			Name:      strings.TrimSpace(rdmDiskCR.Name),
-			Namespace: migrationplan.Namespace,
-		}, rdmDisk)
-		if err != nil {
-			return fmt.Errorf("failed to get RDMDisk CR: %w", err)
-		}
-		// Migration Plan controller only sets ImportToCinder to true,
-		// the RDMDisk controller ensures that RDM disk is managed and
-		// imported to Cinder
-		patch := client.MergeFrom(rdmDisk.DeepCopy())
-		if !rdmDisk.Spec.ImportToCinder {
+
+		// Use retry to handle resourceVersion conflicts gracefully,
+		// on error it will re-fetch the latest version and before
+		// retrying update will check importToCinder flag again.
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			rdmDisk := &vjailbreakv1alpha1.RDMDisk{}
+			if err := r.Get(ctx, types.NamespacedName{
+				Name:      strings.TrimSpace(rdmDiskCR.Name),
+				Namespace: migrationplan.Namespace,
+			}, rdmDisk); err != nil {
+				return err
+			}
+			// Idempotent guard - check if already marked for import
+			if rdmDisk.Spec.ImportToCinder {
+				log.FromContext(ctx).Info("Skipping update, already marked for import",
+					"rdmDisk", rdmDisk.Name)
+				return nil
+			}
+
+			// Migration Plan controller only sets ImportToCinder to
+			// true, the RDMDisk controller ensures that RDM disk is
+			// managed and imported to Cinder
```

Comment:
Is there a delay in marking it as ImportToCinder true?
if there is, is above check condition enough?

Reply:
No, this is just a spec update so that happens instantly. Basically, we say this "should" imported into Cinder. Once that happens we update status.